ride_share.rb

```diff
@@ -0,0 +1,272 @@
+
+# Setup the entire data structure:
+# based off of the notes you have above, create the
+# and manually write in data presented in rides.csv
+# You should be copying and pasting the literal data
+# into this data structure, such as "DR0004"
+# and "3rd Feb 2016" and "RD0022"
+
+rides = [
+    # Ride 0
+    {
+        driver_id: "DR0004", 
+        date: { 
+                    day: "3rd",
+                    month: "Feb",
+                    year: 2016
+                },
+        cost: 5,
+        rider_id: "RD0022",
+        rating: 5
+    },
+    # Ride 1
+    {
+        driver_id: "DR0001", 
+        date: { 
+                    day: "3rd",
+                    month: "Feb",
+                    year: 2016
+                },
+        cost: 10,
+        rider_id: "RD0003",
+        rating: 3
+    },
+    # Ride 2
+    {
+        driver_id: "DR0002",
+        date: {
+            day: "3rd",
+            month: "Feb",
+            year: 2016
+        },
+        cost: 25,
+        rider_id: "RD0073",
+        rating: 5
+    },
+    # Ride 3
+    {
+        driver_id: "DR0001",
+        date: {
+            day: "3rd",
+            month: "Feb",
+            year: 2016
+        },
+        cost: 30,
+        rider_id: "RD0015",
+        rating: 4
+    },
+    # Ride 4
+    {
+        driver_id: "DR0003",
+        date: {
+            day: "4th",
+            month: "Feb",
+            year: 2016
+        },
+        cost: 5,
+        rider_id: "RD0066",
+        rating: 5
+    },
+    # Ride 5
+    {
+        driver_id: "DR0004",
+        date: {
+            day: "4th",
+            month: "Feb",
+            year: 2016
+        },
+        cost: 10,
+        rider_id: "RD0022",
+        rating: 4
+    },
+    # Ride 6
+    {
+        driver_id: "DR0002",
+        date: {
+            day: "4th",
+            month: "Feb",
+            year: 2016
+        },
+        cost: 15,
+        rider_id: "RD0013",
+        rating: 1
+    },
+    # Ride 7
+    {
+        driver_id: "DR0003",
+        date: {
+            day: "5th",
+            month: "Feb",
+            year: 2016
+        },
+        cost: 50,
+        rider_id: "RD0003",
+        rating: 2
+    },
+    # Ride 8
+    {
+        driver_id: "DR0002",
+        date: {
+            day: "5th",
+            month: "Feb",
+            year: 2016
+        },
+        cost: 35,
+        rider_id: "RD0066",
+        rating: 3
+    },
+    # Ride 9
+    {
+        driver_id: "DR0004",
+        date: {
+            day: "5th",
+            month: "Feb",
+            year: 2016
+        },
+        cost: 20,
+        rider_id: "RD0073", 
+        rating: 5
+    },
+    # Ride 10
+    {
+        driver_id: "DR0001",
+        date: {
+            day: "5th",
+            month: "Feb",
+            year: 2016
+        },
+        cost: 45,
+        rider_id: "RD0003", 
+        rating: 2
+    }
+]
+
+# Analysis data structure
+drivers = [
+    # Driver 1
+    {
+        driver_id: "DR0001",
```

Comment:
I like that you establish this data structure to hold the results up front - it makes it very clear what your code is working toward. What if you had to aggregate data from 100 or 100,000 drivers? Could you generate this structure using a loop?